### DISCUSSION TITLE
question about deeph inference

### DISCUSSION THREAD

**Discussion by userchba** [mzjb/DeepH-pack#78] [Q&A] [answered]
I have a question about deep-inference. Because of the time cost, I set 10 data not 400 data which the demo give. Appreciate it if there is anyone can give a helping hand.
####### Begin 5.sparse_calc
./abacus_CNT.json
[ Info: read h5
Time for reading h5: 1.775418996810913s
[ Info: construct sparse matrix in the format of COO
Time for constructing sparse matrix in the format of COO: 1.2740960121154785s
[ Info: convert sparse matrix to the format of CSC
Time for converting to the format of CSC: 0.2988770008087158s
ERROR: LoadError: KeyError: key "which_k" not found
Stacktrace:
 [1] getindex(h::Dict{String, Any}, key::String)
   @ Base ./dict.jl:482
 [2] main()
   @ Main ~/miniconda3/lib/python3.9/site-packages/deeph/inference/sparse_calc.jl:218
 [3] top-level scope
   @ ~/miniconda3/lib/python3.9/site-packages/deeph/inference/sparse_calc.jl:412
in expression starting at /home/hbchen/miniconda3/lib/python3.9/site-packages/deeph/inference/sparse_calc.jl:412
Traceback (most recent call last):
  File "/home/hbchen/miniconda3/bin/deeph-inference", line 8, in <module>
    sys.exit(main())
  File "/home/hbchen/miniconda3/lib/python3.9/site-packages/deeph/scripts/inference.py", line 150, in main
    assert capture_output.returncode == 0
AssertionError

**Comment by userchba** [comment, marked as answer]
I have solved the question with set the parameter in the file abacus_CNT.json

**Reply by Luo-Xiaofeng** [reply]
Can you please tell me how did you modify abacus_CNT.json?I am experiencing the same problem. Thank you!

**Comment by vegaly** [comment]
Hello, I also encountered this problem. How should the which_k parameter be set? In addition, I would like to ask how to obtain the settings of these two parameters in the abacus_CNT.json file:
 "fermi_level": -2.4,
"k_data": ["60 0 0 0 0 0 0.5 Γ Z"]?
Thank you!

**Reply by okihane** [reply]
> Hello, I also encountered this problem. How should the which_k parameter be set? In addition, I would like to ask how to obtain the settings of these two parameters in the abacus_CNT.json file: "fermi_level": -2.4, "k_data": ["60 0 0 0 0 0 0.5 Γ Z"]? Thank you!

you can modify the json file like this,
{
  "calc_job": "band",
  "which_k": 0,
  "max_iter": 300,
  "num_band": 60,
  "fermi_level": -2.4,
  "k_data": ["60 0 0 0 0 0 0.5 Γ Z"]
}
and you can refer to the source code for the meaning of these keys, https://github.com/mzjb/DeepH-pack/blob/141faaaba1d5e17248283ef939bcd2ead2d8e677/deeph/inference/sparse_calc.jl#L218
